src/types.ts

```diff
@@ -1,4 +1,5 @@
 import * as ethers from "ethers";
+import { IDWithClaimStatus } from ".";
```

Comment:
You should not be importing types from `index.ts` into the `types.ts` file.

Define the `IDWithClaimStatus` type in `types.ts`, not in `index.ts`

Types belong in the types file.